### PR TITLE
Some scheduling cleanup and automatic types planning stream opening

### PR DESCRIPTION
**Pull request by jackh726**
A couple cleanups for scheduling, and add job from opening types planning meeting thread and requesting updates.

(I can split the cleanups out if needed)

### CODE REVIEW

File: src/handlers/types_planning_updates.rs
Comment (Mark-Simulacrum) [inline src/handlers/types_planning_updates.rs:107]: I would recommend bailing *for that issue* -- e.g., leaving it in the list always. Debugging failures in triagebot requires infra/triagebot team involvement since this error message will be somewhere in our logs (and/or in the db in this case IIRC), which is going to be painful and means you get *zero* pings out rather than pinging excessively for small number of issues.

You could also consider looking at the last updated date on the issue rather than specifically the update date of the last comment, which doesn't run into this problem.

File: src/handlers/types_planning_updates.rs
Comment (Mark-Simulacrum) [inline src/handlers/types_planning_updates.rs:169]: Can you leave a comment here indicating which stream this is supposed to be?

File: src/handlers/types_planning_updates.rs
Comment (Mark-Simulacrum) [inline src/handlers/types_planning_updates.rs:184]: Can we move this function to `zulip.rs`? That will hopefully help avoiding re-implementing it twice, and feels more consistent with the general code structure (Zulip and GitHub handling are in their own modules, handlers just call that code).

Also: I don't think this function is the right way to do this. We should use the team API and the zulip & github IDs present there, which doesn't rely on this information being available to Zulip (which AFAIK isn't guaranteed). So perhaps we can just delete it for now.

File: src/handlers/types_planning_updates.rs
Comment (jackh726) [reply src/handlers/types_planning_updates.rs:107]: >  e.g., leaving it in the list always

This is a good idea.

> consider looking at the last updated date on the issue

This is a better idea and I'll switch to this.

File: src/handlers/types_planning_updates.rs
Comment (jackh726) [reply src/handlers/types_planning_updates.rs:169]: Yeah, I'll just make a const at the top, since this is the same as the other job too.

File: src/handlers/types_planning_updates.rs
Comment (jackh726) [reply src/handlers/types_planning_updates.rs:184]: So, I mostly copied this code from `zulip.rs` (I think, it was somewhere).

The reason I didn't just use the zulip id is because `Recipient::Private` needs to email. Not sure if this strictly required for the API, or just an artifact of the implementation here.

For now, I'll just remove this function. If I need it in the future, I'll do some related cleanups in `zulip.rs`.

Comment (Mark-Simulacrum) [reply src/handlers/types_planning_updates.rs:184]: Sounds good. Yeah, not sure on email vs. Zulip user ID etc.

**Comment by jackh726** [comment]
@Mark-Simulacrum addressed review and fixed failed cron expression parsing (guess cron doesn't support "last Monday of the month")

**Comment by jackh726** [comment]
(Not that I necessary expect this to be merged hastily, but can we wait until at least Tuesday to merge so we don't accidentally trigger *this* months planning meeting job)

**Review by Mark-Simulacrum — APPROVED** [review]

**Pull request merged**
